src/Components/MatchHistoryComp/MatchHistoryComp.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import axios from 'axios';
+import { getFullUrl , findChampions } from '../../Modules/Helper';
+import { Spinner } from 'react-bootstrap';
+
+class MatchHistoryComp extends React.Component {
+
+    state = {
+        match_history: null
+    }
+
+    componentDidUpdate() {
+        if (this.props.display) {
```

Comment:
This is a weird way to handle component updating. Try to make things more clear with an isVisible prop, and data fetching according to UI actions(i.e. not out of the blue)